mapper/orm/src/main/java/org/hibernate/search/mapper/orm/schema/management/SchemaManagementStrategyName.java

```diff
@@ -0,0 +1,118 @@
+	 * A strategy that does not change indexes nor their schema,
+	 * but checks that indexes exist and validates their schema on startup.
+	 * <p>
+	 * An exception will be thrown on startup if:
+	 * <ul>
+	 *     <li>Indexes are missing</li>
+	 *     <li>OR, with Elasticsearch only, indexes exist but their schema does not match the requirements
+	 *     of the Hibernate Search mapping:
+	 *     missing fields, fields with incorrect type, missing analyzer definitions or normalizer definitions, ...
+	 *     </li>
+	 * </ul>
+	 * <p>
+	 * With Lucene, the schema of indexes is not validated.
+	 */
+	VALIDATE( "validate" ),
+
+	/**
+	 * A strategy that creates missing indexes and their schema on startup,
+	 * but does not touch existing indexes and assumes their schema is correct without validating it.
+	 * <p>
+	 * Note that creating indexes and their schema will not populate the indexed data:
+	 * a newly created index will always be empty.
+	 * To populate indexes with pre-existing data, use mass indexing.
+	 */
+	CREATE( "create" ),
+
+	/**
+	 * A strategy that creates missing indexes and their schema on startup,
+	 * and updates the schema of existing indexes if possible.
+	 * <p>
+	 * Note that creating indexes or updating their schema will not populate or update the indexed data:
+	 * newly created indexes will always be empty.
+	 * To populate indexes with pre-existing data, use mass indexing.
+	 * <p>
+	 * <strong>Warning:</strong> This strategy is unfit for production environments,
+	 * due to the limitations explained below.
+	 * It should only be relied upon during development,
+	 * to easily add new fields to an existing index.
+	 * <p>
+	 * <strong>Warning:</strong> with Elasticsearch, if analyzer/normalizer definitions have to be updated, the index will be closed automatically during the update.
+	 * <p>
+	 * <strong>Warning:</strong> with Elasticsearch, many scenarios can cause schema updates to fail:
+	 * a field changed its type from string to integer, an analyzer definition changed, ...
+	 * In such cases, the only workaround is to drop and re-create the index.
+	 */
+	CREATE_OR_UPDATE( "create-or-update" ),
+
+	/**
+	 * A strategy that drops existing indexes and re-creates them and their schema on startup.
+	 * <p>
+	 * Note that dropping indexes means losing all indexed data,
+	 * and creating indexes will not populate them:
+	 * the newly created indexes will always be empty.
+	 * To populate indexes with pre-existing data, use mass indexing.
+	 */
+	DROP_AND_CREATE( "drop-and-create" ),
+
+	/**
+	 * A strategy that drops existing indexes and re-creates them and their schema on startup,
+	 * then drops the indexes on shutdown.
+	 * <p>
+	 * Note that dropping indexes means losing all indexed data,
+	 * and creating indexes will not populate them:
+	 * the newly created indexes will always be empty.
+	 * To populate indexes with pre-existing data, use mass indexing.
+	 */
+	DROP_AND_CREATE_AND_DROP( "drop-and-create-and-drop" );
```

Comment:
I know some users have a very small index and they re-create it on every startup of their application... which is why I left out this part: it can make sense in a production environment.

Reply:
You're right these are indexes not the main data!